core/src/main/java/jenkins/views/PartialHeader.java

```diff
@@ -0,0 +1,46 @@
+package jenkins.views;
+
+import java.util.stream.Collectors;
+
+import org.apache.log4j.Logger;
+
+import hudson.ExtensionList;
+import hudson.init.InitMilestone;
+import hudson.init.Initializer;
+import hudson.util.AdministrativeError;
+import jenkins.model.Jenkins;
+
+/**
+ * {@link Header} that relies on core resources (images, CSS, JS, etc.) to perform
+ * partial replacements.
+ * 
+ * Given this kind of header is not independent, compatibility should be managed by the
+ * specific {@link Header} compatibility header version value
+ * 
+ * @see Header
+ */
+public abstract class PartialHeader extends Header {
+
+    private static Logger LOGGER = Logger.getLogger(PartialHeader.class.getName());
```

Comment:
Should be using jul, not log4j.

Reply:
Updated in https://github.com/jenkinsci/jenkins/pull/5909/commits/2fa8ea7fb63537b56c48cf390782423004ddf634